ibis/expr/analysis.py

```diff
@@ -774,7 +774,7 @@ def _check_fusion(self, root):
 
             # a * projection
             if (isinstance(val, ir.TableExpr) and
-                (self.parent.op().equals(val.op()) or
+                (self.parent.op().compatible_with(val.op()) or
```

Comment:
Compatible_with is nice, but we seriously need to refactor the analysis code.

Reply:
yeah it's getting out of hand